zcash_primitives/src/note_encryption.rs

```diff
@@ -483,15 +483,15 @@ pub fn try_sapling_compact_note_decryption<P: consensus::Parameters>(
 
 /// Recovery of the full note plaintext by the sender.
 ///
-/// Attempts to decrypt and validate the given `enc_ciphertext` using the given `ovk`.
+/// Attempts to decrypt and validate the given `enc_ciphertext` using the given `ock`.
 /// If successful, the corresponding Sapling note and memo are returned, along with the
 /// `PaymentAddress` to which the note was sent.
 ///
-/// Implements section 4.17.3 of the Zcash Protocol Specification.
-pub fn try_sapling_output_recovery<P: consensus::Parameters>(
+/// Implements part of section 4.17.3 of the Zcash Protocol Specification.
+/// for decryption using a Full Viewing Key see [try_sapling_output_recovery].
+pub fn try_sapling_output_recovery_with_ock<P: consensus::Parameters>(
     height: u32,
-    ovk: &OutgoingViewingKey,
-    cv: &edwards::Point<Bls12, Unknown>,
+    ock: &[u8],
```

Comment:
`ock` is not something I'm inclined to expose as a standalone object. However, this API is required for implementing shielded-output-specific recovery, and would be necessary for an external implementation of payment disclosures. Once we have specified Sapling payment disclosures, and implemented them in the Rust crates, we may revisit this and remove it from the public API.